torchrl/envs/mb_envs/dreamer.py

```diff
@@ -0,0 +1,258 @@
+from math import sqrt
+
+import torch
+import torch.distributions as d
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.distributions.kl import kl_divergence
+
+from torchrl.data import TensorDict
+from torchrl.modules.tensordict_module import TensorDictModule, TensorDictSequence
+from ..model_based import ModelBasedEnv
+
+
+class DreamerEnv(ModelBasedEnv):
+    def __init__(
+        self,
+        obs_depth=32,
+        rssm_hidden=200,
+        rnn_hidden_dim=200,
+        state_dim=20,
+        device="cpu",
+        dtype=None,
+        batch_size=None,
+    ):
+        super().__init__(
+            world_model=DreamerModel(
+                obs_depth=obs_depth,
+                rssm_hidden=rssm_hidden,
+                rnn_hidden_dim=rnn_hidden_dim,
+                state_dim=state_dim,
+            ),
+            reward_model=TensorDictModule(
```

Comment:
I would move these out of the call to `__init__` for clarity

Reply:
What do you mean by this? MBEnv requires to have a world model and a reward as an init. And you cant declare nn.Module before running __init__

Reply:
I just meant
No 
```
fun(a=makestuff(...), b=makeotherstuff(...))
```

But
```

a=makestuff(...)
b=makeotherstuff(...)

fun(a, b)
```